blockchain-agent/src/main/java/com/sphereon/alfresco/blockchain/agent/tasks/AlfrescoRepository.java

```diff
@@ -0,0 +1,170 @@
+import com.alfresco.apis.model.SearchRequest;
+import com.google.common.collect.ImmutableMap;
+import com.sphereon.alfresco.blockchain.agent.model.AlfrescoBlockchainRegistrationState;
+import com.sphereon.libs.blockchain.commons.Digest;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.time.OffsetDateTime;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class AlfrescoRepository {
+    private static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ISO_OFFSET_DATE_TIME;
+    private static final org.slf4j.Logger logger = LoggerFactory.getLogger(AlfrescoRepository.class);
+
+    private static final String EXCEPTION_MESSAGE_FETCH_CONTENT = "An error occurred whilst fetching content: ";
+    private static final String EXCEPTION_MESSAGE_GET_NODE = "An error occurred whilst getting node %s: %s";
+
+    private final SearchApi alfrescoSearchApi;
+    private final ApiClient cmisApiClient;
+    private final NodesApi alfrescoNodesApi;
+
+    private String model;
+    private String registrationStateProperty;
+
+    public AlfrescoRepository(final SearchApi alfrescoSearchApi,
+                              final ApiClient cmisApiClient,
+                              final NodesApi alfrescoNodesApi,
+                              @Value("${sphereon.blockchain.agent.alfresco.query.model}") final String model,
+                              @Value("${sphereon.blockchain.agent.alfresco.query.registration-state:RegistrationState}") final String registrationStateProperty) {
+        this.alfrescoSearchApi = alfrescoSearchApi;
+        this.cmisApiClient = cmisApiClient;
+        this.alfrescoNodesApi = alfrescoNodesApi;
+        this.model = model;
+        this.registrationStateProperty = registrationStateProperty;
+    }
+
+    public List<ResultSetRowEntry> selectEntries(final AlfrescoBlockchainRegistrationState state) throws ApiException {
+        final SearchRequest request = new SearchRequest();
+        final RequestQuery query = new RequestQuery();
+        query.setLanguage(RequestQuery.LanguageEnum.AFTS);
+        query.setQuery('{' + model + '}' + registrationStateProperty + ':' + state.getKey());
+        request.setQuery(query);
+        RequestInclude include = new RequestInclude();
+        include.add("properties");
+        request.include(include);
+
+        final ResultSetPaging pagedResult = alfrescoSearchApi.search(request);
+        return pagedResult.getList().getEntries();
+    }
+
+    public List<NodeEntry> selectEntries(final List<String> alfrescoNodeIds) {
```

Comment:
The Entry/Entries names of these methods leave a bit to be desired I think

Reply:
Good point. I refactored the argument names, but left the method names out.

Reply:
Changed it to `selectAlfrescoNodes` for now. It's an Alfresco repository though, so we can leave out the Alfresco part, but `selectNodes` is also a bit generic. Not sure if `selectAlfrescoDocumentNodes` is semantically correct.